Marlin/src/feature/tmc_util.cpp

```diff
@@ -995,6 +999,45 @@
   }
   void tmc_disable_stallguard(TMC2660Stepper, const bool) {};
 
+  #if ENABLED(SPI_ENDSTOPS)
+
+    bool Endstops::tmc_spi_homing_check() {
+      bool hit = false;
+      #if X_SPI_SENSORLESS
+        if (TEST(tmc_spi_homing, X_AXIS) && stepperX.test_stall_status()) {
+          SBI(live_state, X_STOP);
+          hit = true;
+        }
+      #endif
+      #if Y_SPI_SENSORLESS
+        if (TEST(tmc_spi_homing, Y_AXIS) && stepperY.test_stall_status()) {
+          SBI(live_state, Y_STOP);
+          hit = true;
+        }
+      #endif
+      #if Z_SPI_SENSORLESS
+        if (TEST(tmc_spi_homing, Z_AXIS) && stepperZ.test_stall_status()) {
+          SBI(live_state, Z_STOP);
+          hit = true;
+        }
+      #endif
+      return hit;
+    }
+
+    void Endstops::clear_endstop_state() {
+      #if X_SPI_SENSORLESS
+        CBI(live_state, X_STOP);
+      #endif
+      #if Y_SPI_SENSORLESS
+        CBI(live_state, Y_STOP);
+      #endif
+      #if Z_SPI_SENSORLESS // Needs hardware testing
+        CBI(live_state, Z_STOP);
+      #endif
+    }
```

Comment:
I suppose this could just be `live_state = 0` but that might interfere with real endstops.